### PR TITLE
Bert tutorial

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Add tutorial for finetuning with bert

### Essentials ###
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/examples/sentence_embedding/bert.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+```{.python .input}
+# create a dataloader with batch size 32
+bert_dataloader = mx.gluon.data.DataLoader(data_train, batch_size=32, shuffle=True, last_batch='rollover')
+
+trainer = gluon.Trainer(model.collect_params(), 'adam', {'learning_rate': 5e-6, 'epsilon': 1e-9})
+params = []
+for p in model.collect_params().values():
+    if p.grad_req != 'null':
+        params.append(p)
```
Comment (szha) [inline docs/examples/sentence_embedding/bert.md]: list comprehension can be used here.

File: docs/examples/sentence_embedding/bert.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+in improving many downstream tasks, they require task-specific architectures.
+Devlin, Jacob, et al proposed BERT [1] (Bidirectional Encoder Representations
+from Transformers), which **fine-tunes** deep bidirectional representations on a
+wide range of tasks with minimal task-specific parameters, and obtained state-
+of-the-art results.
+
+In this tutorial, we will focus on fine-tuning with the
+pre-trained BERT model to classify semantically equivalent sentence pairs.
+Specifically, we will:
+1. load the state-of-the-art pre-trained BERT model.
```
Comment (szha) [inline docs/examples/sentence_embedding/bert.md]: I had the impression that there needs to be an empty line before the first list item to make the formatting correct. Maybe not necessary for md but you can double check the rendered results.

File: docs/examples/sentence_embedding/bert.md (hunk 1 above)
Comment (szha) [reply docs/examples/sentence_embedding/bert.md]: Also maybe explain what grad req does

File: docs/examples/sentence_embedding/bert.md (hunk 2 above)
Comment (eric-haibin-lin) [reply docs/examples/sentence_embedding/bert.md]: line added

File: docs/examples/sentence_embedding/bert.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    metric.reset()
+    step_loss = 0
+    for batch_id, (token_ids, valid_length, segment_ids, label) in enumerate(bert_dataloader):
+        with mx.autograd.record():
+            
+            # load data to GPU
+            token_ids = token_ids.as_in_context(mx.gpu())
+            valid_length = valid_length.as_in_context(mx.gpu())
+            segment_ids = segment_ids.as_in_context(mx.gpu())
+            label = label.as_in_context(mx.gpu())
```
[the remark this answers is not in the document]
Comment (szhengac) [reply docs/examples/sentence_embedding/bert.md]: ctx has been defined in ### Set Environment.

File: docs/examples/sentence_embedding/bert.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
+# Fine-tuning Sentence Pair Classification with BERT
+
+Pre-trained language
+representations have been shown to improve many downstream NLP tasks such as
+question-answering, and natural language inference. To apply pre-trained
```
Comment (sxjscience) [inline docs/examples/sentence_embedding/bert.md]: Should be “question answering”.

**Comment by mli** [comment]
Job PR-431/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-431/9/index.html

**Comment by mli** [comment]
Job PR-431/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-431/10/index.html

**Comment by mli** [comment]
Job PR-431/11 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-431/11/index.html

**Comment by fierceX** [comment]
I looked at GLUONNLP's Bert pre-training model, found that there is no Chinese model, only a multi-language model, whether you can make a Chinese single language pre-training model. In addition, Gluon's pre-training is from the tensorflow of the conversion or re-training to get. If the conversion can provide conversion code, if it is retraining can provide training script?

**Comment by szha** [comment]
@fierceX thanks for asking. It's encouraging to see immediate attention after BERT code merge!

Right now we are focusing on pushing the existing code out as 0.5.0 (hopefully today!) so we might not be able to get to it immediately. The models are converted so feel free to connect with Haibin on the conversion code. We will be looking into including more models after the 0.5.0 release. It would be awesome if you could help out on some of the conversions 💯

**Comment by eric-haibin-lin** [comment]
@fierceX It is converted from the original TF model. I should have made it clear in the doc though. Thanks for asking! I can PR the Chinese model and the code for the conversion in 2 - 3 days.
